src/scenechange/mod.rs

```diff
@@ -301,24 +295,29 @@ impl<T: Pixel> SceneChangeDetector<T> {
   ) -> ScenecutResult {
     // Downscaling both frames for comparison
     // Moving scaled frames to buffer
-    if self.frame_buffer.is_empty() {
-      let frame1_scaled = frame1.planes[0].downscale(self.scale_factor);
-      self.frame_buffer.push(frame1_scaled);
-
-      let frame2_scaled = frame2.planes[0].downscale(self.scale_factor);
-      self.frame_buffer.push(frame2_scaled);
+    if let Some(frame_buffer) = self.frame_buffer.as_deref_mut() {
+      frame_buffer.swap(0, 1);
+      frame2.planes[0]
+        .downscale_in_place(self.scale_factor, &mut frame_buffer[1]);
     } else {
-      self.frame_buffer.remove(0);
-      self.frame_buffer.push(frame2.planes[0].downscale(self.scale_factor));
+      self.frame_buffer = Some(Box::new([
+        frame1.planes[0].downscale(self.scale_factor),
+        frame2.planes[0].downscale(self.scale_factor),
+      ]));
     }
 
-    let delta =
-      self.delta_in_planes(&self.frame_buffer[0], &self.frame_buffer[1]);
+    if let Some(frame_buffer) = self.frame_buffer.as_deref() {
+      let delta = self.delta_in_planes(&frame_buffer[0], &frame_buffer[1]);
 
-    ScenecutResult {
-      intra_cost: self.threshold as f64,
-      threshold: self.threshold as f64,
-      inter_cost: delta as f64,
+      ScenecutResult {
+        intra_cost: self.threshold as f64,
+        threshold: self.threshold as f64,
+        inter_cost: delta as f64,
+      }
+    } else {
+      // SAFETY: `self.frame_buffer` was just initialized in this function,
+      // and must be a `Some` variant
+      unsafe { std::hint::unreachable_unchecked() }
```

Comment:
There shouldn't be that much overhead from get() compared to the other tasks of this function (downscaling).

Reply:
I changed this to use `unreachable!()`, so it should be fine now.